File: Cpp_vs_Python/Procedures_Functions/Procedures_Functions.py
# =====================================================================================
# Procedures_Functions.py
# =====================================================================================

def say_hello(count: int, message: str) -> None:
    for _ in range(count):
        print(message)
    print()


def example_procedures_functions_01() -> None:
    say_hello(2, "I'm saying Hello :)")
    say_hello(3, "I'm saying Hello :)")


def power_of(base: int, exponent: int) -> int:
    result = 1
    for _ in range(exponent):
        result *= base
    return result


def example_procedures_functions_02() -> None:
    base = 2
    exponent = 5
    
    result = power_of(base, exponent)
    print(f"{base} to the power of {exponent}: {result}")


# Aufruf der Beispiel-Funktionen wie im C++ Code
if __name__ == "__main__":
    example_procedures_functions_01()
    example_procedures_functions_02()


# =====================================================================================
# End-of-File
# =====================================================================================

File: Cpp_vs_Python/Procedures_Functions/Procedures_Functions.cpp
// =====================================================================================
// Procedures_Functions.cpp
// =====================================================================================

#include <cstddef>
#include <print>
#include <string>

// procedure
static void sayHello(std::size_t count, const std::string& message)
{
    for (std::size_t i{}; i != count; ++i) {
        std::println("{}", message);
    }
    std::println();
}

static void example_procedures_functions_01()
{
    sayHello(2, "I'm saying Hello :)");
    sayHello(3, "... and Goodbye!");
}

static std::size_t powerOf(std::size_t base, std::size_t exponent)
{
    std::size_t result{ 1 };

    for (std::size_t i{}; i != exponent; ++i) {
        result *= base;
    }

    return result;
}

static void example_procedures_functions_02()
{
    std::size_t base{ 2 };
    std::size_t exponent{ 5 };
    std::size_t result{};

    result = powerOf(base, exponent);
    std::println("{} to the power of {}: {}", base, exponent, result);
}

void example_procedures_functions()
{
    example_procedures_functions_01();
    example_procedures_functions_02();
}

// =====================================================================================
// End-of-File
// =====================================================================================
